drivers/mspi/mspi_stm32_xspi.c

```diff
@@ -0,0 +1,1896 @@
+		return -EIO;
+	}
+	__HAL_LINKDMA(&dev_data->hmspi.xspi, hdmarx, dev_data->hdma_rx);
+
+	return 0;
+}
+
+/**
+ * API implementation of mspi_config : controller configuration.
+ *
+ * @param spec Pointer to MSPI device tree spec.
+ * @return 0 if successful.
+ * @return -Error if fail.
+ */
+static int mspi_stm32_xspi_config(const struct mspi_dt_spec *spec)
+{
+	int ret;
+	const struct mspi_cfg *config = &spec->config;
+	const struct mspi_stm32_conf *dev_cfg = spec->bus->config;
+	struct mspi_stm32_data *dev_data = spec->bus->data;
+	uint32_t ahb_clock_freq;
+
+	ret = mspi_validate_config(config);
+	if (ret != 0) {
+		return ret;
+	}
+
+	(void)pm_device_runtime_get(spec->bus);
+	/* Prevent the clocks to be stopped during the request */
+	pm_policy_state_lock_get(PM_STATE_SUSPEND_TO_IDLE, PM_ALL_SUBSTATES);
+
+	/* pinctrl */
+	ret = pinctrl_apply_state(dev_cfg->pcfg, PINCTRL_STATE_DEFAULT);
+	if (ret < 0) {
+		LOG_ERR("MSPI pinctrl setup failed");
+		goto end;
+	}
+
+	if (dev_data->dev_cfg.dqs_enable && !dev_cfg->mspicfg.dqs_support) {
+		LOG_ERR("MSPI dqs mismatch (not supported but enabled)");
+		ret = -ENOTSUP;
+		goto end;
+	}
+
+	dev_cfg->irq_config();
+
+	ret = mspi_stm32_xspi_activate(spec->bus);
+	if (ret != 0) {
+		goto end;
+	}
+
+	ret = mspi_config_clocks(dev_cfg, &ahb_clock_freq);
+	if (ret != 0) {
+		goto end;
+	}
+
+	ret = mspi_hal_init(dev_cfg, dev_data, ahb_clock_freq);
+	if (ret != 0) {
+		goto end;
+	}
+
+#if defined(HAL_XSPIM_IOPORT_1) || defined(HAL_XSPIM_IOPORT_2)
+	/* XSPI I/O manager config */
+	XSPIM_CfgTypeDef mspi_mgr_cfg;
+
+	if (dev_data->hmspi.xspi.Instance == XSPI1) {
+		mspi_mgr_cfg.IOPort = HAL_XSPIM_IOPORT_1;
+	}
+	if (dev_data->hmspi.xspi.Instance == XSPI2) {
+		mspi_mgr_cfg.IOPort = HAL_XSPIM_IOPORT_2;
+	}
+	mspi_mgr_cfg.nCSOverride = HAL_XSPI_CSSEL_OVR_DISABLED;
+	mspi_mgr_cfg.Req2AckTime = 1;
+	if (HAL_XSPIM_Config(&dev_data->hmspi.xspi, &mspi_mgr_cfg,
+			     HAL_XSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
+		LOG_ERR("XSPI M config failed");
+		ret = -EIO;
+		goto end;
+	}
+#endif
+
+#if defined(DLYB_XSPI1) || defined(DLYB_XSPI2) || defined(DLYB_OCTOSPI1) || defined(DLYB_OCTOSPI2)
+	HAL_XSPI_DLYB_CfgTypeDef mspi_delay_block_cfg = {0};
+	(void)HAL_XSPI_DLYB_GetClockPeriod(&dev_data->hmspi.xspi, &mspi_delay_block_cfg);
+	mspi_delay_block_cfg.PhaseSel /= 4;
+	if (HAL_XSPI_DLYB_SetConfig(&dev_data->hmspi.xspi, &mspi_delay_block_cfg) != HAL_OK) {
+		LOG_ERR("XSPI DelayBlock failed");
+		ret = -EIO;
+		goto end;
+	}
+	LOG_DBG("Delay Block Init");
+#endif
+
+#ifdef CONFIG_MSPI_DMA
+	ret = mspi_dma_setup(dev_cfg, dev_data);
+	if (ret != 0) {
+		goto end;
+	}
+#endif
+
+	if (k_sem_count_get(&dev_data->ctx.lock) == 0) {
+		k_sem_give(&dev_data->ctx.lock);
+	}
+	if (config->re_init) {
+		k_mutex_unlock(&dev_data->lock);
+	}
+
+end:
+	pm_policy_state_lock_put(PM_STATE_SUSPEND_TO_IDLE, PM_ALL_SUBSTATES);
+	(void)pm_device_runtime_put(spec->bus);
+
+	if (ret == 0) {
+		LOG_INF("MSPI config'd");
```

Comment:
```suggestion
		LOG_INF("MSPI configured");
```
Does this really belong to the info log level? What about `LOG_DBG()`?